core/liblwm2m.h

```diff
@@ -496,20 +507,39 @@ typedef enum
 typedef uint8_t lwm2m_binding_t;
 
 /*
- * LWM2M block1 data
+ * LWM2M block data
  *
- * Temporary data needed to handle block1 request.
- * Currently support only one block1 request by server.
+ * Temporary data needed to handle block1 request and block2 responses.
  */
-typedef struct _lwm2m_block1_data_ lwm2m_block1_data_t;
+typedef enum
+{
+    BLOCK_1,
+    BLOCK_2,
+} block_type_t;
+
+typedef union _block_data_identifier_
+{
+    char * uri;                               // resource string if block1 
```

Comment:
Rename, all other uri members are of type lwm2m_uri_t

Reply:
The char * is a used as an unique identifier constructed from both the uri and query parameters, there is no need to handle the uri or query parameters as structs and will in fact make it more complicated in this case. No changes